aws/rust-runtime/aws-config/src/credential_process.rs

```diff
@@ -11,33 +11,95 @@ use aws_smithy_types::date_time::Format;
 use aws_smithy_types::DateTime;
 use aws_types::credentials::{future, CredentialsError, ProvideCredentials};
 use aws_types::{credentials, Credentials};
-use std::borrow::Cow;
 use std::fmt;
 use std::process::Command;
 use std::time::SystemTime;
 
-/// Credentials Provider
-pub struct CredentialProcessProvider {
-    command: String,
-}
+pub(crate) struct CommandWithSensitiveArgs<T>(T);
 
-/// Returns the given `command` string with arguments redacted if there were any
-pub(crate) fn debug_fmt_command_string(command: &str) -> Cow<'_, str> {
-    match command.find(char::is_whitespace) {
-        Some(index) => Cow::Owned(format!("{} ** arguments redacted **", &command[0..index])),
-        None => Cow::Borrowed(command),
+impl<T> CommandWithSensitiveArgs<T>
+where
+    T: AsRef<str>,
+{
+    pub(crate) fn new(value: T) -> Self {
+        Self(value)
+    }
+
+    pub(crate) fn unredacted(&self) -> &str {
+        self.0.as_ref()
     }
 }
 
-impl fmt::Debug for CredentialProcessProvider {
+impl<T> fmt::Display for CommandWithSensitiveArgs<T>
+where
+    T: AsRef<str>,
+{
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         // Security: The arguments for command must be redacted since they can be sensitive
-        f.debug_struct("CredentialProcessProvider")
-            .field("command", &debug_fmt_command_string(&self.command))
-            .finish()
+        let command = self.0.as_ref();
+        match command.find(char::is_whitespace) {
+            Some(index) => write!(f, "{} ** arguments redacted **", &command[0..index]),
+            None => write!(f, "{}", command),
+        }
+    }
+}
+
+impl<T> fmt::Debug for CommandWithSensitiveArgs<T>
+where
+    T: AsRef<str>,
+{
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{:?}", format!("{}", self))
     }
 }
 
+impl<T> Clone for CommandWithSensitiveArgs<T>
```

Comment:
I think this is just `#[derive(Clone)]`?

Reply:
It needs the `where T: Clone` clause, but maybe I can just put that bound on the struct itself and then derive...

Reply:
If I take that approach, then each other trait `impl` needs to have a `T: Clone` bound added, so I don't think it's much of a savings.